Source/WebCore/dom/Document.cpp

```diff
@@ -10723,19 +10725,51 @@ void Document::flushDeferredRenderingIsSuppressedForViewTransitionChanges()
     }
 }
 
-RefPtr<ViewTransition> Document::startViewTransition(RefPtr<ViewTransitionUpdateCallback>&& updateCallback)
+ExceptionOr<RefPtr<ViewTransition>> Document::startViewTransition(StartViewTransitionCallbackOptions&& callbackOptions)
 {
     if (!globalObject())
         return nullptr;
 
-    Ref viewTransition = ViewTransition::create(*this, WTFMove(updateCallback));
+    if (callbackOptions && std::holds_alternative<StartViewTransitionOptions>(*callbackOptions) && !settings().viewTransitionsTypeEnabled())
+        return Exception { ExceptionCode::TypeError };
+
+    // https://drafts.csswg.org/css-view-transitions-2/#extend-document-types
+
+    // 1. Let updateCallback be null.
+    RefPtr<ViewTransitionUpdateCallback> updateCallback = nullptr;
+
+    Vector<AtomString> activeTypes { };
+
+    // 2. If callbackOptions is an an UpdateCallback, set updateCallback to callbackOptions.
+    // 3. Otherwise, if callbackOptions is a StartViewTransitionOptions, then set updateCallback to callbackOptions’s update.
+    // Also extract active types for step 6.
+    if (callbackOptions) {
+        WTF::switchOn(*callbackOptions, [&](const RefPtr<JSViewTransitionUpdateCallback>& argUpdateCallback) {
+            updateCallback = WTFMove(argUpdateCallback);
+        }, [&](const StartViewTransitionOptions& transitionOptions) {
+            ASSERT(settings().viewTransitionsTypeEnabled());
+
+            updateCallback = WTFMove(transitionOptions.update);
+            if (transitionOptions.types)
+                activeTypes = WTFMove(*transitionOptions.types);
+        });
+    }
+
+    // 4. If this’s active view transition is not null and its outbound post-capture steps is not null, then:
+    // FIXME: assume "outbound post-capture steps" is null for now. This is needed to do cross-document view transitions.
+
+    // 5. Let viewTransition be the result of running the method steps for startViewTransition(updateCallback) given updateCallback.
+    // https://drafts.csswg.org/css-view-transitions-1/#ViewTransition-prepare
+
+    Ref viewTransition = ViewTransition::create(*this, WTFMove(updateCallback), WTFMove(activeTypes));
 
     if (RefPtr activeViewTransition = m_activeViewTransition)
         activeViewTransition->skipViewTransition(Exception { ExceptionCode::AbortError, "Old view transition aborted by new view transition."_s });
 
     setActiveViewTransition(WTFMove(viewTransition));
     scheduleRenderingUpdate(RenderingUpdateStep::PerformPendingViewTransitions);
-    return m_activeViewTransition;
+
+    return { m_activeViewTransition.copyRef() };
```

Comment:
Why this change?

Reply:
It's needed to return `ExceptionOr<RefPtr<...>>` (or I'm wrong and there's a different way). Anyway, now that we don't need to raise an exception, this isn't needed anymore.